node/podcontroller.go

```diff
@@ -158,6 +163,8 @@ type PodControllerConfig struct {
 	ConfigMapInformer corev1informers.ConfigMapInformer
 	SecretInformer    corev1informers.SecretInformer
 	ServiceInformer   corev1informers.ServiceInformer
+
+	EnvResolver podutils.ResolverFunc
```

Comment:
Since we are already resolving envs, I think we don't need to make it configurable, and instead make the non-pod informers optional... e.g. if a configmap informer is not provided, then we don't resolve config maps.... or maybe it's all or nothing... provide all these informers and we'll resolve env vars, otherwise we won't.

Reply:
> Since we are already resolving envs, I think we don't need to make it configurable, and instead make the non-pod informers optional... e.g. if a configmap informer is not provided, then we don't resolve config maps.... or maybe it's all or nothing... provide all these informers and we'll resolve env vars, otherwise we won't.

Agree with @cpuguy83 , actually we just want to the same behavior with `kubelet`. Since the current solution can satisfy the requirement, I think that is enough.


@justnoise WDYT?